zivid_camera/test/test_zivid_camera.cpp

```diff
@@ -335,6 +349,84 @@ TEST_F(ZividNodeTest, testDynamicReconfigureNodesAreAvailable)
   ASSERT_FALSE(ros::service::waitForService(prefix + "frame_11/set_parameters", wait_duration));
 }
 
+TEST_F(ZividNodeTest, testCapture2D)
+{
+  waitForReady();
+
+  std::optional<sensor_msgs::CameraInfo> color_camera_info;
+  std::optional<sensor_msgs::Image> image;
+  auto color_camera_info_sub =
+      subscribe<sensor_msgs::CameraInfo>(color_camera_info_topic_name, [&](const auto& r) { color_camera_info = *r; });
+  auto color_image_color_sub =
+      subscribe<sensor_msgs::Image>(color_image_color_topic_name, [&](const auto& i) { image = *i; });
+
+  auto assert_num_topics_received = [&](std::size_t numTopics) {
+    ASSERT_EQ(color_camera_info_sub.numMessages(), numTopics);
+    ASSERT_EQ(color_image_color_sub.numMessages(), numTopics);
+  };
+
+  sleepAndSpin(ros::Duration(1));
+  assert_num_topics_received(0);
+
+  // Capture fails when no frames are enabled
+  zivid_camera::Capture2D capture;
+  ASSERT_FALSE(ros::service::call(capture_2d_service_name, capture));
+  sleepAndSpin(ros::Duration(1));
+  assert_num_topics_received(0);
+
+  enableFirst2DFrame();
+  ASSERT_TRUE(ros::service::call(capture_2d_service_name, capture));
+  sleepAndSpin(ros::Duration(1));
+  assert_num_topics_received(1);
+
+  auto verifyImageAndCameraInfo = [this](const auto& img, const auto& info) {
+    assertCameraInfoForFileCamera(info);
+
+    ASSERT_EQ(img.width, 1920U);
+    ASSERT_EQ(img.height, 1200U);
+    ASSERT_EQ(img.step, 4U * 1920U);
```

Comment:
Suggestion: give a name to the number 4 here.

Reply:
can do